onboard/catalog_actions.go

```diff
@@ -18,9 +18,9 @@ const (
 )
 
 type RepoActions interface {
-	ApplyImport(ctx context.Context, it Iterator, dryRun bool) (*InventoryImportStats, error)
+	ApplyImport(ctx context.Context, it Iterator, stats *Stats, dryRun bool) error
```

Comment:
separate the callback from the stats.
go back to returning the stats.
the callback should be passed here and called from ApplyImport, perhaps with a dedicated object or perhaps with the stats.

Reply:
also, instead of callback consider a progress-bar-style interface